plugins/uefi-capsule/fu-uefi-capsule-device.c

```diff
@@ -677,9 +677,15 @@ fu_uefi_capsule_device_prepare_firmware(FuDevice *device,
 	if (!fu_firmware_parse_stream(firmware, stream, 0x0, flags, error))
 		return NULL;
 	if (sz_reqd == 0) {
-		g_info("required ESP free space is not configured, using 2 x %uMB + 20MB",
-		       (guint)fu_firmware_get_size(firmware) / (1024 * 1024));
-		sz_reqd = fu_firmware_get_size(firmware) * 2 + (20u * 1024 * 1024);
+		if (fu_device_has_private_flag(FU_DEVICE(self),
+					       FU_UEFI_CAPSULE_DEVICE_FLAG_NO_EXTRA_ESP_SPACE)) {
+			g_info("no additional ESP free space required");
+			sz_reqd = fu_firmware_get_size(firmware);
```

Comment:
do we need extra space for the FWUPD EFI executable?

Reply:
Yup, that's valid. On my system it's 68K, so lets add an extra 200KB just to be on the safe side.